### PR TITLE
Add burn in YASPool

### PR DESCRIPTION
**Pull request by dpinones**
Closes: #91

### CODE REVIEW

**Review by rcatalan98 — CHANGES_REQUESTED** [review]

File: crates/yas_core/src/contracts/yas_pool.cairo [hunk 1]
```diff
@@ -663,7 +730,7 @@ mod YASPool {
         /// @return amount1 the amount of token1 owed to the pool, negative if the pool should pay the recipient
         fn modify_position(
             ref self: ContractState, params: ModifyPositionParams
-        ) -> (Info, i256, i256) // TODO: noDelegateCall
+        ) -> (PositionInfo, i256, i256) // TODO: noDelegateCall
```
Comment (rcatalan98) [inline crates/yas_core/src/contracts/yas_pool.cairo:733]:
```suggestion
        ) -> (PositionInfo, i256, i256)
```

File: crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo [hunk 2]
```diff
@@ -1520,6 +1520,176 @@ mod YASPoolTests {
...
+
+            let info_position = yas_pool
+                .get_position(
+                    PositionKey { owner: OTHER(), tick_lower: min_tick, tick_upper: max_tick }
+                );
+
+            assert(info_position.liquidity == 0, 'wrong liquidity');
+            assert(info_position.tokens_owed_0 != 0, 'wrong tokens_owed_0');
+            assert(info_position.tokens_owed_1 != 0, 'wrong tokens_owed_1');
+        // TODO: Uncomment when the swap has tests
```
[the remark this answers is not in the document]
Comment (dpinones) [reply crates/yas_core/src/tests/test_contracts/test_yas_pool.cairo:1616]: ✔️

File: crates/yas_core/src/contracts/yas_pool.cairo [hunk 3]
```diff
@@ -558,6 +580,51 @@ mod YASPool {
             self.unlock();
             (amount_0, amount_1)
         }
+
+        /// @inheritdoc IUniswapV3PoolActions
+        /// @dev noDelegateCall is applied indirectly via _modifyPosition
```
[the remark this answers is not in the document]
Comment (dpinones) [reply crates/yas_core/src/contracts/yas_pool.cairo:585]: ✔️

File: crates/yas_core/src/contracts/yas_pool.cairo (hunk 1 above)
Comment (dpinones) [reply crates/yas_core/src/contracts/yas_pool.cairo:733]: ✔️

**Review by dubzn — COMMENTED** [review]
Seems good to me! But I've been reviewing, and there are several test cases missing that are in UniswapV3Pool.spec.ts and the burn() function is used. I'm not sure if you plan to add them later, or it would be good to include them directly in this PR. What do you think?

in the `describe('#mint', () => { ..` theres some tests, can we see it later in a meet?

**Review by dubzn — APPROVED** [review]

**Review by uri-99 — APPROVED** [review]

**Comment by rcatalan98** [comment]
For the future we will need to remove the ToDo in the tests to be sure the behavior is the correct one.

**Pull request merged**